product/models.py

```diff
@@ -29,13 +29,9 @@ def save_product(self):
         try:
             Product.full_clean(self)
         except ValidationError as e:
-            self.delete_product(self.qr_code)
+            self.delete()
             raise e
         return self
 
-    @staticmethod
-    def delete_product(prd_qr):
-        try:
-            Product.filter_qr(prd_qr).delete()
-        except Exception:
-            pass
+    def __str__(self) -> str:
```

Comment:
Is this function relevant to this PR?

Reply:
It is a super function that i overrides for debuging,it can be in any pr because each object has it

Reply:
I agree that this should not be a part on this PR. If you want it added then you need a different PR for that (which also includes tests for that function).

Reply:
Fix the above and we can merge this PR.

Reply:
fixed